ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListSourceExecutor.java

```diff
@@ -208,13 +263,50 @@ private static SourceDescriptionWithWarnings describeSource(
         SourceDescriptionFactory.create(
             dataSource,
             extended,
-            getQueries(ksqlEngine, q -> q.getSourceNames().contains(dataSource.getName())),
-            getQueries(ksqlEngine, q -> q.getSinkName().equals(dataSource.getName())),
-            topicDescription
+            sourceQueries,
+            sinkQueries,
+            topicDescriptionOptional,
+            sourceConsumerOffsets
         )
     );
   }
 
+  private static List<SourceConsumerGroupOffset> consumerOffsets(
+      final TopicDescription topicDescription,
+      final Map<TopicPartition, ListOffsetsResultInfo> topicAndStartOffsets,
+      final Map<TopicPartition, ListOffsetsResultInfo> topicAndEndOffsets,
+      final Map<TopicPartition, OffsetAndMetadata> topicAndConsumerOffsets
+  ) {
+    final List<SourceConsumerGroupOffset> sourceConsumerGroupOffsets = new ArrayList<>();
+    for (TopicPartitionInfo topicPartitionInfo : topicDescription.partitions()) {
+      final TopicPartition tp = new TopicPartition(topicDescription.name(),
+          topicPartitionInfo.partition());
+      final ListOffsetsResultInfo startOffsetResultInfo = topicAndStartOffsets.get(tp);
+      final ListOffsetsResultInfo endOffsetResultInfo = topicAndEndOffsets.get(tp);
+      final OffsetAndMetadata offsetAndMetadata = topicAndConsumerOffsets.get(tp);
+      sourceConsumerGroupOffsets.add(
+          new SourceConsumerGroupOffset(
+              topicPartitionInfo.partition(),
+              startOffsetResultInfo != null ? startOffsetResultInfo.offset() : 0,
+              endOffsetResultInfo != null ? endOffsetResultInfo.offset() : 0,
+              offsetAndMetadata != null ? offsetAndMetadata.offset() : 0
```

Comment:
when will these be `null` and is `0` a good default?

Reply:
It will be `null` when no offsets has been stored yet. e.g. stream created a new topic but topic does not have any data yet.

`0` seemed like a good default, though it might not be the right value: it gives the impression that offset `0` has been committed.

Would `-1` or just a hyphen `-` be better representation for this scenario?

Also, instead of dealing with this here, I can wrap this as an optional an let the client decide how to print. wdyt?